pandas/_libs/writers.pyx

```diff
@@ -17,6 +12,10 @@ ctypedef fused pandas_string:
     bytes
 
 
+cdef extern from 'Python.h':
+    Py_ssize_t PyUnicode_GET_LENGTH(object o) nogil
```

Comment:
Hmm didn't realize this wasn't in Cython. I guess for now stick with `PyUnicode_GET_SIZE` which is there though will be removed in Python 4.0 in favor of this - can deal with that migration comprehensively in our code base later

Reply:
> can deal with that migration comprehensively in our code base later

Good idea +1, probably will make it easier to review this PR. Done!